sdk/file_sdk/XMPFileFormat/xmp_file_format.cpp

```diff
@@ -1,15 +1,169 @@
 #include "xmp_file_format.h"
 
+#include "xmp_stream.h"
+#include <XMP.incl_cpp>
+#include <XMP.hpp>
+#include <time.h>
+#include <string_utils.h>
+#include <mutex>
+
+using std::string;
+using std::vector;
+
 namespace mip {
 namespace file {
 
+const XMP_StringPtr kMsipNamespace = "http://www.microsoft.com/msip/1.0/";
+const XMP_StringPtr kMsipLabelNamespace = "http://www.microsoft.com/msip/label/1.0/";
+const XMP_StringPtr kMsipLabelExtendedNamespace = "http://www.microsoft.com/msip/label/extended/1.0/";
+const XMP_StringPtr kLabelsArrayName = "labels";
+const XMP_StringPtr kExtendedStructName = "extended";
+
+bool Initialized = false;
+std::mutex InitMutex;
+
+static string GetStringByKey(_In_ const SXMPMeta* metadata, _In_ const string &labelsItemPath, _In_ const XMP_StringPtr fieldName, _In_ const XMP_StringPtr alternativeFieldName)
+{
+  std::string path, value;
+  SXMPUtils::ComposeStructFieldPath(kMsipNamespace, labelsItemPath.c_str(), kMsipLabelNamespace, fieldName, &path);
+  metadata->GetProperty(kMsipNamespace, path.c_str(), &value, 0);
+  if (value.empty() && alternativeFieldName != nullptr)
+  {
+    SXMPUtils::ComposeStructFieldPath(kMsipNamespace, labelsItemPath.c_str(), kMsipLabelNamespace, alternativeFieldName, &path);
+    metadata->GetProperty(kMsipNamespace, path.c_str(), &value, 0);
+  }
+
+  return value;
+}
+
+static vector<Tag> Deserialize(_In_ const SXMPMeta* metadata)
```

Comment:
why static?

Reply:
No reason

---
In reply to: [136424354](https://github.com/AzureAD/rms-sdk-for-cpp/pull/140#discussion_r136424354) [](ancestors = 136424354)